src/troute-network/troute/AbstractNetwork.py

```diff
@@ -281,3 +436,423 @@ def astype(self, type, columns=None):
+            if restart_parameters.get("start_datetime", None):
+                t0_str = restart_parameters.get("start_datetime")
+                
+                def _try_parsing_date(text):
+                    for fmt in (
+                        "%Y-%m-%d_%H:%M", 
+                        "%Y-%m-%d_%H:%M:%S", 
+                        "%Y-%m-%d %H:%M", 
+                        "%Y-%m-%d %H:%M:%S", 
+                        "%Y/%m/%d %H:%M", 
+                        "%Y/%m/%d %H:%M:%S"
+                    ):
+                        try:
+                            return datetime.strptime(text, fmt)
+                        except ValueError:
+                            pass
+                    LOG.error('No valid date format found for start_datetime input. Please use format YYYY-MM-DD_HH:MM')
+                    quit()
+                    
+                self._t0 = _try_parsing_date(t0_str)
+            
+            else:
+                raise(RuntimeError("No start_datetime provided in config file for cold start."))
+
+        LOG.debug(
+            "channel initial states complete in %s seconds."\
+            % (time.time() - start_time)
+        )
+
+    def build_forcing_sets(self,):
+
+        forcing_parameters = self.forcing_parameters
+        supernetwork_parameters = self.supernetwork_parameters
+
+        run_sets           = forcing_parameters.get("qlat_forcing_sets", None)
+        qlat_input_folder  = forcing_parameters.get("qlat_input_folder", None)
+        nts                = forcing_parameters.get("nts", None)
+        max_loop_size      = forcing_parameters.get("max_loop_size", 12)
+        dt                 = forcing_parameters.get("dt", None)
+
+        geo_file_type      = supernetwork_parameters.get('geo_file_type')
+
+        try:
+            qlat_input_folder = pathlib.Path(qlat_input_folder)
+            assert qlat_input_folder.is_dir() == True
+        except TypeError:
+            raise TypeError("Aborting simulation because no qlat_input_folder is specified in the forcing_parameters section of the .yaml control file.") from None
+        except AssertionError:
+            raise AssertionError("Aborting simulation because the qlat_input_folder:", qlat_input_folder,"does not exist. Please check the the nexus_input_folder variable is correctly entered in the .yaml control file") from None
+
+        forcing_glob_filter = forcing_parameters.get("qlat_file_pattern_filter", "*.NEXOUT")
+
+        if forcing_glob_filter=="nex-*":
+            print("Reformating qlat nexus files as hourly binary files...")
+            binary_folder = forcing_parameters.get('binary_nexus_file_folder', None)
+            qlat_files = qlat_input_folder.glob(forcing_glob_filter)
+
+            #Check that directory/files specified will work
+            if not binary_folder:
+                raise(RuntimeError("No output binary qlat folder supplied in config"))
+            elif not os.path.exists(binary_folder):
```

Comment:
There are three `os.path` occurrences in this file, those can be moved to `Pathlib` and the os import removed for consistency.  Not high priority for this PR to be merged, but worth an issue and a small subsequent PR to clean this up.